.github/workflows/verify.yml

```diff
@@ -13,65 +13,75 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-name: Go
-# name: verify TODO rename (requires repo settings update)
-
+name: Check Unit and Integration tests
 # yamllint disable-line rule:truthy
 on:
   push:
-    branches: [master]
+    branches:
+      - master
   pull_request:
-    branches: [master, issue-99-cosmos-upgrade-v0.44]
+    branches:
+      - master
+      - issue-99-cosmos-upgrade-v0.44
   workflow_dispatch:
 
 jobs:
-
-  build:
-    name: Build  # TODO remove (requires repo settings update)
+  changes:
     runs-on: ubuntu-latest
+    outputs:
+      go: ${{ steps.filter.outputs.go }}
+      src: ${{ steps.filter.outputs.src }}
     steps:
-
-      - name: Build
-        run: true
-
-  unit:
-    name: Unit tests  # TODO remove (requires repo settings update)
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        if: ${{ !env.ACT }}
+        id: filter
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          filters: |
+            go:
+              - '**/*.go'
+              - 'go.mod'
+              - 'go.sum'
+            src:
+              - '!**.md'
+              - '!.github/*'
+  unit-tests:
+    name: Check unit tests
     runs-on: ubuntu-latest
+    needs:
+      - lint
     steps:
-
-      - name: Set up Go 1.x
-        uses: actions/setup-go@v2
+      - uses: actions/setup-go@v2
         with:
           go-version: ^1.16
-
-      - name: Check out code into the Go module directory
-        uses: actions/checkout@v2
-
+      - uses: actions/checkout@v3
       - uses: actions/cache@v2
+        if: ${{ !env.ACT }}
         with:
           path: |
             ~/.cache/go-build
             ~/go/pkg/mod
           key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
           restore-keys: |
             ${{ runner.os }}-go-
-
-      - name: Run units
-        run: make test
-
-  integration:
-    name: Integration tests  # remove (requires repo settings update)
+      - uses: haveyoudebuggedit/gotestfmt-action@v2
+      - name: Run tests
+        run: |
+          set -euo pipefail
+          # shellcheck disable=SC2046
+          go test -json -v $(go list ./... | grep -v '/integration_tests') 2>&1 | tee /tmp/gotest.log | gotestfmt
+  integration-tests:
+    if: needs.changes.outputs.src == 'true'
+    name: Integration tests
     runs-on: ubuntu-latest
+    needs:
+      - changes
     steps:
-
-      - name: Set up Go 1.x
-        uses: actions/setup-go@v2
+      - uses: actions/setup-go@v2
         with:
-          go-version: ^1.16
-
-      - name: Check out code into the Go module directory
-        uses: actions/checkout@v2
-
+          go-version: ^1.13
```

Comment:
Any reasons to downgrade Go version here from 1.16 to 1.13?

Reply:
I missed [this](https://github.com/zigbee-alliance/distributed-compliance-ledger/commit/8dbd9f9726e888bc986a21024f8070a29a187028) upgrade